synapse/storage/databases/main/registration.py

```diff
@@ -220,26 +224,43 @@ async def set_renewal_token_for_user(
         await self.db_pool.simple_update_one(
             table="account_validity",
             keyvalues={"user_id": user_id},
-            updatevalues={"renewal_token": renewal_token},
+            updatevalues={"renewal_token": renewal_token, "token_used_ts_ms": None},
             desc="set_renewal_token_for_user",
         )
 
-    async def get_user_from_renewal_token(self, renewal_token: str) -> str:
-        """Get a user ID from a renewal token.
+    async def get_user_from_renewal_token(
+        self, renewal_token: str
+    ) -> Tuple[str, bool, int, Optional[int]]:
+        """Get a user ID and renewal status from a renewal token.
 
         Args:
             renewal_token: The renewal token to perform the lookup with.
 
         Returns:
-            The ID of the user to which the token belongs.
+            A tuple of containing the following values:
+                * The ID of a user to which the token belongs.
+                * Whether an email has been sent to the user with the token, and
+                    that they haven't renewed their account with it yet.
```

Comment:
Why would an email not have been sent? Is this return value just whether they've renewed or not?

Reply:
Err, changing this comment was part of using the `email_sent` boolean to determine whether the current token had been used to renew the account. With switching to `token_used_ts_ms` this comment doesn't really need to be updated anymore.

This bool would be false if the user had renewed their account, and an email has not been sent since then. Otherwise it is set to true upon an email being sent out.

In the grand scheme of things we probably could have just renamed `email_sent` to `token_unused`, but that doesn't feel as clean as having a timestamp of when the user renewed their account with the given token.

Reply:
Oh, we're not even using `email_sent` anymore huh. I'll remove this from the return tuple.